addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/handler/LoxoneMiniserverHandler.java

```diff
@@ -355,12 +437,39 @@ public void dispose() {
         }
     }
 
+    /**
+     * Create and add a new channel to the channels list.
+     *
+     * @param channels
+     *            list of channels to add the channel to
+     * @param itemType
+     *            item type for the channel
+     * @param typeId
+     *            channel type ID for the channel
+     * @param channelId
+     *            channel ID
+     * @param channelLabel
+     *            channel label
+     * @param channelDescription
+     *            channel description
+     * @param tags
+     *            tags for the channel or null if no tags needed
+     * @return
+     *         true if channel was created and added to the list
+     */
     private boolean addChannel(List<Channel> channels, String itemType, ChannelTypeUID typeId, ChannelUID channelId,
             String channelLabel, String channelDescription, Set<String> tags) {
         if (channelId != null && itemType != null && typeId != null && channelDescription != null) {
-            channels.add(ChannelBuilder.create(channelId, itemType).withType(typeId).withLabel(channelLabel)
-                    .withDescription(channelDescription + " : " + channelLabel).withDefaultTags(tags).build());
-            return true;
+            ChannelBuilder builder = ChannelBuilder.create(channelId, itemType).withType(typeId).withLabel(channelLabel)
+                    .withDescription(channelDescription + " : " + channelLabel);
+            if (tags != null) {
+                builder = builder.withDefaultTags(tags);
+            }
+            Channel newChannel = builder.build();
+            if (channels != null) {
```

Comment:
if channels  == null the prior effort seems unneeded too. Can't the check be part of the first if statement?